wrapper/cpp/main.go

```diff
@@ -332,6 +598,80 @@ func KeyvaultGetPublicKeyCV(serverName *C.char, groupName *C.char, keyvaultName
+//   - errMessageSize: the size of pre allocated errMessageBuffer. If the actual error message size is bigger, the error message will be truncated to errMessageSize
+// return:
+//   - int: the win32 code. If output buffer is not big enough this function will return Win32ErrorInsufficientBuffer which is win32 error ERROR_INSUFFICIENT_BUFFER.
+//     Caller when seeing this error should allocate outputBuffer to size outputBufferSize
+// Calling pattern:
+//   call KeyvaultGetPublicKeyV2 for the first time. If returned error Win32ErrorInsufficientBuffer, caller allocate outputBuffer to size outputBufferSize and call KeyvaultGetPublicKeyV2 again.
+//
+//export KeyvaultGetPublicKeyV2
+func KeyvaultGetPublicKeyV2(serverName *C.char, groupName *C.char, keyvaultName *C.char, keyName *C.char, cv *C.char, timeoutInSeconds C.int, outputBuffer *C.char, outputBufferSize *C.ulonglong, errMessageBuffer *C.char, errMessageSize C.ulonglong) C.int {
+	if (serverName == nil || groupName == nil || keyvaultName == nil || keyName == nil || outputBufferSize == nil || cv == nil) {
+		telemetry.EmitWrapperTelemetry("KeyvaultGetPublicKeyV2", C.GoString(cv), "", "InvalidArgument", C.GoString(serverName))
+		copyErrorMessage(C.CString("Invalid Argument"), errMessageBuffer, errMessageSize)
+		return C.int(Win32ErrorBadArg)
+	}
+
+	keyClient, err := getKeyvaultKeyClient(C.GoString(serverName), C.GoString(cv))
+	if err != nil {
+		telemetry.EmitWrapperTelemetry("KeyvaultGetPublicKeyV2", C.GoString(cv), err.Error(), "getKeyvaultKeyClient", C.GoString(serverName))
+		copyErrorMessage(C.CString(telemetry.FilterSensitiveData(err.Error())), errMessageBuffer, errMessageSize)
+		return C.int(Win32ErrorFunctionFail)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeoutInSeconds)*time.Second)
+	defer cancel()
+
+	keys, err := keyClient.Get(ctx, C.GoString(groupName), C.GoString(keyvaultName), C.GoString(keyName))
+	if err != nil {
+		telemetry.EmitWrapperTelemetry("KeyvaultGetPublicKeyV2", C.GoString(cv), err.Error(), "keyClient.Get", C.GoString(serverName))
+		copyErrorMessage(C.CString(telemetry.FilterSensitiveData(err.Error())), errMessageBuffer, errMessageSize)
+		return C.int(Win32ErrorFunctionFail)
+	}
+
+	if keys == nil || len(*keys) <= 0 {
+		telemetry.EmitWrapperTelemetry("KeyvaultGetPublicKeyV2", C.GoString(cv), "", "EmptyKey", C.GoString(serverName))
+		copyErrorMessage(C.CString("Returned key is empty"), errMessageBuffer, errMessageSize)
+		return C.int(Win32ErrorFunctionFail)
+	}
+
+	pemPkcs1KeyPub := (*keys)[0].Value
+	return handleOutputBuffer(serverName, *pemPkcs1KeyPub, cv, outputBuffer, outputBufferSize, "KeyvaultGetPublicKeyV2")
+}
+
+func handleOutputBuffer(serverName *C.char, resultGoString string, cv *C.char, outputBuffer *C.char, outputBufferSize *C.ulonglong, funcName string) C.int {

```

Comment:
>handleOutputBuffer(s [](http://example.com/codeflow?start=5&length=20)

Uber nit: Consider moving this to the top with the other helpers and add comment.